calico_containers/pycalico/util.py

```diff
@@ -90,4 +90,24 @@ def get_hostname():
     except KeyError:
         # The user does not have a set $HOSTNAME. Since this is a common
         # scenario, return socekt.gethostname instead of just erroring.
-        return socket.gethostname()
\ No newline at end of file
+        return socket.gethostname()
+
+def check_ports(port_list):
+    """
+    Checks whether a list of ports are within range of 0 and 65535.
+    The port list may contains ints or strings including a a range of ports,
+    delimited by a colon.
+
+    :param port_list:
+    :return: a Boolean: True if in range, False if not in range
+    """
+    in_range = True
+    for port in port_list:
+        if ":" in str(port):
+            in_range = check_ports(port.split(":"))
+        else:
+            in_range = 0 <= int(port) < 65536
```

Comment:
this will raise an exception if the port value is not a number.  That may be correct, but should make that clear in the doc string.